### DISCUSSION TITLE
How to check if another controller "is granted" for the logged-in user?

### DISCUSSION THREAD

**Discussion by D3strukt0r** [symfony/symfony#54182] [Q&A]
e.g. I have a Show Controller for my Categories, and there is an "actions" property I want to provide the frontend with the list of actions that the current logged-in user can do on this entity (e.g. `["edit", "delete"]`).

In the backend I want to simply give a list of Controllers (since I use `__invoke`, just passing the class name should be enough: e.g. `["edit" => CategoryEditController::class, "delete" => CategoryDeleteController::class]`) to a function which tells me whether that controller is accessible for the user or not (e.g. `["edit" => true, "delete" => false]`)

I haven't figured out yet how I can check another class's `IsGranted` which is the crucial issue I'm facing. Does anyone have an idea?

Also need to consider, IsGranted could not just be a string but an expression and also contain a subject. The `security.yaml` could also have constraints not directly represented in the IsGranted.

**Comment by MatTheCat** [comment]
It could be a good idea to provide an API around your authorization rules (like [EasyAdminBundle’s “permissions”](https://symfony.com/bundles/EasyAdminBundle/current/security.html)), else you’d need to either duplicate them, or send/forward an actual request to said controllers..!

**Comment by stof** [comment]
This is more complex than that actually. The `IsGranted` annotation can vote on a subject, which will generally be one of the controller argument (the advanced case supports an expression). The access is not for a controller. It is for a controller with a given set of arguments.

There is no built-in way to perform those checks.
